doc/man7/password-handling.pod

```diff
@@ -0,0 +1,130 @@
+=pod
+
+=head1 NAME
+
+password handling - How diverse parts of OpenSSL treat pass phrases
+
+=head1 DESCRIPTION
+
+In a modern world with all sorts of character encodings, the treatment of pass
+phrases has become increasingly complex.
+This manual page attempts to give an overview over how this problem is
+currently addressed in different parts of the openssl library.
+
+=head2 The general case
+
+The OpenSSL library doesn't treat pass phrases in any special way as a general
+rule, and trusts the application or user to choose a suitable character set
+and stick to that throughout the life of affected objects.
+This means that for an object that was encrypted using a pass phrase encoded in
+ISO-8859-1, that object needs to be decrypted using a pass phrase encoded in
+ISO-8859-1.
+Anything else will have the decryption fail.
+
+=head2 PKCS#12
+
+PKCS#12 is a bit different regarding pass phrase encoding.
+The standard stipulates that the pass phrase shall be encoded as a BMPString,
+i.e. UCS-2 BE.
+
+OpenSSL tries to adapt to this requirements in one of the following manners:
+
+=over 4
+
+=item 1.
+
+Tries to read the pass phrase as UTF-8 and re-encode that to UTF-16 (which is
+the same as UCS-2 for characters U+0000 to U+D7FF and U+E000 to U+FFFF, but
+becomes an expansion for any other character), or failing that, proceeds with
```

Comment:
The reading is not what fails and causes the fallback to step 2, but rather the attempted use of the reencoded passphrase as a decryption key.  So maybe "or if decryption fails with that reencoding"?

Reply:
Not quite, it's actually so that if input appears as malformed UTF-8, it falls back to 1.0.x way *without* attempting to decipher protected object. Though it does another try if decryption fails. Decryption failure after second try would mean that object is in fact protected with UTF-8 string that user *failed* to provide (on Linux most likely because non-UTF-8 locale was used, and on Windows because environment variable wasn't set).